### DISCUSSION TITLE
Cancel My Push Action

### DISCUSSION THREAD

**Discussion by Dumte** [community/community#54437] [Repositories]
### Select Topic Area

Bug

### Body

![Screenshot (15)](https://user-images.githubusercontent.com/61066433/236194757-f85c25ff-8f2c-40b8-89de-3bce7e24a45e.png)
![Screenshot (16)](https://user-images.githubusercontent.com/61066433/236194838-bdbd4e2d-8968-4f0c-8144-d9f8c9e67877.png)

I got into this mess and need help.

I wanted to push my files to github and got stuck with network uploading problem.

Now I want to revert all the changes and cancel all my uploads.

**May I also ask for a better and efficient way to push my files to GitHub from my local folders on Visual Studio Code Editor.**

Help me please

**Comment by ccalvocm** [comment]
Greetings @Dumte 

Here are some steps you can follow to revert your changes and cancel your uploads:

1. Open the terminal in Visual Studio Code by going to View -> Terminal or by pressing Ctrl + Shift + `.
2. Navigate to the local repository where you made the changes by using the cd command.
3. Use the git log command to find the commit hash of the commit you want to revert to.
4. Use the git reset --hard <commit hash> command to revert to the commit you want to go back to. This will discard all changes made after that commit.
5. Use the git push --force command to force push the changes to GitHub and overwrite the remote repository with your local repository.


As for a better and more efficient way to push your files to GitHub from Visual Studio Code, you can use the built-in Git integration. Here are the steps to do so:

1. Open the repository in Visual Studio Code.
2. Make your changes to the files.
3. Open the Source Control panel by going to View -> SCM or by pressing Ctrl + Shift + G.
4. Stage your changes by clicking the "+" icon next to the changed files in the Source Control panel.
5. Add a commit message by typing it in the text box at the top of the Source Control panel and pressing Ctrl + Enter.
6. Push your changes to GitHub by clicking the "..." icon in the Source Control panel and selecting "Push".
7. This should push your changes to GitHub without any issues. Let me know if you have any further questions or concerns.
